src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java

```diff
@@ -558,4 +559,16 @@ private List<DimensionAttribute> generateAttributes() {
     return list;
   }
 
+  @Test
+  public void testGetterNotExists() {
```

Comment:
AFAICT, while valuable to test previously existing behavior, this test does not address the new feature being added: that of being able to have fields and getters/setters from base classes. It would be good to add more tests covering the new feature. Can be on a separate PR, if there is need to merge this sooner.